src/main/java/com/app/bookstore/repository/impl/BookRepositoryImpl.java

```diff
@@ -33,7 +29,7 @@ public Book save(Book book) {
             if (transaction != null) {
                 transaction.rollback();
             }
-            throw new HibernateException("Can't save the book: " + book, e);
+            throw new EntityNotFoundException("Can't save the book: " + book, e);
```

Comment:
I think that this exception it's not accurate for this case, better create a new DataProcessingException for example